ndcube/utils/cube.py

```diff
@@ -0,0 +1,351 @@
+            # of items yet to be deleted.
+            n = len(dep_pix_axes) - 1  # last element index for converting iterator to index.
+            for j, axes in enumerate(dep_pix_axes[::-1]):
+                if axes.issubset(dpa):
+                    world_axes = np.delete(world_axes, n-j)
+                    world_n_dep_dim = np.delete(world_n_dep_dim, n-j)
+                    dep_pix_axes = np.delete(dep_pix_axes, n-j)
+            # Collect the world axes which share the same set of pixel axes.
+            # To avoid duplication in future iterations of this loop,
+            # remove those world axes from current-level list.
+            # Iterate backwards to delete last items first so as not to change index
+            # of items yet to be deleted.
+            dep_world_axes_i = []
+            n = len(dep_pix_axes_i) - 1  # last element index for converting iterator to index.
+            for j, axes in enumerate(dep_pix_axes_i[::-1]):
+                if axes == dpa:
+                    dep_world_axes_i.append(world_axes_i[n-j])
+                    world_axes_i = np.delete(world_axes_i, n-j)
+                    dep_pix_axes_i = np.delete(dep_pix_axes_i, n-j)
+            # We started this function by defining the corners for 1D axes.
+            # This means that to fully describe the corners, the number of world axes
+            # we need is the number of shared pixel axes minus 1.
+            # If there are more, discard the excess.
+            ii = i - 1
+            dep_world_axes_i = np.array(dep_world_axes_i)[:ii]
+            # For i number of shared pixel coords, we need 2^i corners.
+            # Add the required number of corners by duplicating current one
+            # then edit them so they contain the correct world values.
+            # First double the number of corners.
+            corners = np.repeat(corners[np.newaxis], 2, axis=0)
+            # Then multiple the number of corners of ii (i-1) since we
+            # started off by defining two corners.
+            corners = np.repeat(corners[np.newaxis], ii, axis=0)
+            # Fill in the correct world values for the new corners.
+            for j in range(ii):
+                corners[j, 0, :, dep_world_axes_i] = lower_corner_values[dep_world_axes_i]
+                corners[j, 1, :, dep_world_axes_i] = upper_corner_values[dep_world_axes_i]
+            # Reshape corners array so it will be predictable for next iteration.
+            corners = corners.reshape((np.prod(corners.shape[:3]), corners.shape[-1]))
+
+    # Reattach units if were present in input.
+    corners = tuple(tuple(corner * corner_unit if corner_unit else corner
+                          for corner, corner_unit in zip(corners[i], corner_units))
+                    for i in range(corners.shape[0]))
+    return corners
+
+
+def get_crop_item_from_points(*world_points_values, wcs, data_shape):
```

Comment:
Consider moving this back onto `NDCube` once `get_crop_item` and `get_crop_item_values` are moved back.